scripts/release.sh

```diff
@@ -148,7 +148,7 @@ main() {
     fi
 
     # Tag release.
-    if [ "$(git tag --list | grep -c "${RELEASE_VERSION}")" -gt 0 ]; then
+    if git tag --list | grep --quiet "^${RELEASE_VERSION}$"; then
```

Comment:
same comment as https://github.com/etcd-io/etcd/pull/19954#discussion_r2092454407 cc @ivanvc @jmhbnz but it's up to your two.